### PR TITLE
[13.0][ADD] stock_reserve_sale_mrp

### PR DESCRIPTION
**Pull request by JaumeBforgeFlow**
Based on `stock_reserve_sale` to cover the case where the product has a KIT BOM

### CODE REVIEW

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM 👍🏽

**Review by JordiMForgeFlow — APPROVED** [review]
Code review

File: stock_reserve_sale_mrp/wizard/sale_stock_reserve.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        self.ensure_one()
+        lines = self.env["sale.order.line"].browse(line_ids)
+        lines_without_kit = self.env["sale.order.line"]
+        vals_list = []
+        for line in lines:
+            bom_kit = self.env["mrp.bom"]._bom_find(
+                product=line.product_id,
+                company_id=line.company_id.id,
+                bom_type="phantom",
+            )
```
Comment (AaronHForgeFlow) [inline stock_reserve_sale_mrp/wizard/sale_stock_reserve.py:34]: Is it possible to use the BOM from the stock move? just wondering if the stock moves are already created.

```suggestion
            bom_kit =  line.move_ids.filtered(lambda m: m.state != 'cancel').mapped('bom_line_id.bom_id').filtered(lambda b: b.type == 'phantom')
```

Comment (JaumeBforgeFlow) [reply stock_reserve_sale_mrp/wizard/sale_stock_reserve.py:34]: Hi @AaronHForgeFlow, at this point the moves are not created. The aim of this module and `stock_reserve_sale` is to allow reservations on the quotation state of the SO.

Comment (AaronHForgeFlow) [reply stock_reserve_sale_mrp/wizard/sale_stock_reserve.py:34]: ok! thanks for letting me know!

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM!

**Review by JoanSForgeFlow — APPROVED** [review]

**Pull request merged**